system-tests/lib/cre/don/don.go

```diff
@@ -82,20 +84,25 @@ func BuildTopology(nodeSetInput []*cretypes.CapabilitiesAwareNodeSet) (*cretypes
 				Value: nodeType,
 			})
 
-			// TODO this will only work with Docker, for CRIB we need a different approach
-			// that will need to be aware of namespace name and node naming pattern
-			host := fmt.Sprintf("%s-node%d", donMetadata.Name, j)
+			// TODO think whether it would make sense for infraDetails to also hold functions that resolve hostnames for various infra and node types
+			// and use it with some default, so that we can easily modify it with little effort
+			host := infra.Host(j, nodeType, donMetadata.Name, infraDetails)
 
-			if nodeSetInput[i].GatewayNodeIndex != -1 && j == nodeSetInput[i].GatewayNodeIndex {
+			if slices.Contains(nodeSetInput[i].DONTypes, cretypes.GatewayDON) && nodeSetInput[i].GatewayNodeIndex != -1 && j == nodeSetInput[i].GatewayNodeIndex {
```

Comment:
ugly, but we cannot use `flags.HasFlags` as flags are only created by this very `BuildTopology` funcion